### PR TITLE
Added resource model

### PR DESCRIPTION
**Pull request by LiorRey**
Relating to #36, solves resource part from #12

Added an SQLAlchemy table/class for handling the data of a resource of a course.

### CODE REVIEW

**Review by mosesnagar — CHANGES_REQUESTED** [review]

File: tutor/models/resource.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+class Resource(db.Model):
+    __tablename__ = 'resource'
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String(100), nullable=False)
+    date_posted = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    content = db.Column(db.Text, nullable=False)
+    link = db.Column(db.Text)
+    user_id = db.Column(db.Integer, db.ForeignKey(user.id), nullable=False)
+    course_id = db.Column(db.Integer, db.ForeignKey(course.id), nullable=False)
```
[the remark this answers is not in the document]
Comment (LiorRey) [reply tutor/models/resource.py:11]: > The `db. ForeignKey` parameter should be a string, isn't it?

at line 10:
It should be the Primary Key of the specific user who posted the resource. And if the id field of the user was declared as an Integer - ForeignKey should be an Integer.

at line 11:
It should be the Primary Key of the specific course for which the resource was posted - which I declared as an Integer, hence ForeignKey is also an Integer.

[the remark this answers is not in the document]
Comment (mosesnagar) [reply tutor/models/resource.py:11]: For line 10: we have no users yet.

For line 11: Agree, but there is no `course` object(or dict) to get `id` property(or key) from. Taken from SQLAlchemy docs: 
image
But if you checked it and there are no problems I think it's fine.

**Comment by ifireball** [comment]
Is this a dependency of PR #45 ? Its not mentioned there...

**Comment by bardielle** [comment]
@LiorRey What about that PR?

**Comment by LiorRey** [comment]
> @LiorRey What about that PR?

**Pull request closed without merging**